processing/db_handler.py

```diff
@@ -134,6 +134,113 @@ def get_base_badm_update_zip_count(self, conn):
             count = cursor.fetchone().get('count')
         return count
 
+    def get_input_files(self, process_id):
+        input_files = set()
+        with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+            query = SQL('SELECT format_qaqc_id '
+                        'FROM qaqc.aggregate_processing_log '
+                        'WHERE data_qaqc_id = %s')
+            cursor.execute(query, (process_id,))
+            for row in cursor:
+                pid = row.get('format_qaqc_id')
+                input_files.add(pid)
+        return input_files
+
+    # def get_sites_with_updates(self):
+    #     site_ids = {}
+    #     with conn.cursor(cursor_factory=RealDictCursor) as cursor:
+    #         query = SQL('SELECT p.site_id, p.process_id, p.base_version '
+    #                     'FROM '
+    #                     '(SELECT proc.site_id, pub.process_id, '
+    #                     'pub.base_version, pub.log_timestamp '
+    #                     'FROM qaqc.publishing_log pub '
+    #                     'INNER JOIN qaqc.processing_log proc '
+    #                     'ON pub.process_id = proc.log_id) p '
+    #                     'INNER JOIN ('
+    #                     'SELECT site_id, pub.process_id, '
+    #                     'MAX(pub.log_timestamp) AS latest_ts '
+    #                     'FROM qaqc.publishing_log pub '
+    #                     'INNER JOIN qaqc.processing_log proc '
+    #                     'ON pub.process_id = proc.log_id '
+    #                     'GROUP BY site_id, pub.process_id, '
+    #                     'pub.log_timestamp) latest_publish '
+    #                     'ON p.log_timestamp = latest_publish.latest_ts '
+    #                     'latest_publish.site_id = p.site_id')
+    #         cursor.execute(query)
+    #         for row in cursor:
+    #             site_id = row.get('site_id')
+    #             site_ids[site_id] = (
+    #                 row.get('base_version'), row.get('process_id'))
+    #     return site_ids
+
+    def define_base_candidates_query(
+            self, pre_query, post_query, state_ids):
+        full_query_components = [pre_query]
+        query_criteria = SQL('s.state_log = %s ')
+        for idx, state_id in enumerate(state_ids):
+            if idx > 0:
+                full_query_components.append(SQL( 'OR '))
```

Comment:
For state_id elements after the first one, looks like only 'OR' is being added. The query criteria also needs to be added.